src/auto-evo/simulation/SimulationCache.cs

```diff
@@ -379,44 +379,73 @@ public void Clear()
         cachedStorageScores.Clear();
     }
 
-    public (float PilusScore, float OxytoxyScore, float MucilageScore) GetPredationToolsRawScores(
+    public (float PilusScore, float OxytoxyScore, float SlimeJetScore) GetPredationToolsRawScores(
         MicrobeSpecies microbeSpecies)
     {
         if (cachedPredationToolsRawScores.TryGetValue(microbeSpecies, out var cached))
             return cached;
 
         var pilusScore = 0.0f;
         var oxytoxyScore = 0.0f;
-        var mucilageScore = 0.0f;
+        var slimeJetScore = 0.0f;
 
         var organelles = microbeSpecies.Organelles.Organelles;
         var organelleCount = organelles.Count;
+        var slimeJetsCount = 0;
+        var topPosition = 0;
+        var bottomPosition = 0;
+        List<int> slimeJetsHorizontalPositions = new List<int>();
 
         for (int i = 0; i < organelleCount; ++i)
         {
             var organelle = organelles[i];
 
+            // TODO: Is there better option for calculating/storing cells boundaries?
+            topPosition = organelle.Position.R < topPosition ? organelle.Position.R : topPosition;
+            bottomPosition = organelle.Position.R > bottomPosition ? organelle.Position.R : bottomPosition;
+
             if (organelle.Definition.HasPilusComponent)
             {
                 pilusScore += Constants.AUTO_EVO_PILUS_PREDATION_SCORE;
                 continue;
             }
 
+            if (organelle.Definition.HasSlimeJetComponent)
+            {
+                slimeJetsHorizontalPositions.Add(organelle.Position.R);
+                ++slimeJetsCount;
+                slimeJetScore += Constants.AUTO_EVO_MUCILAGE_PREDATION_SCORE;
+            }
+
             foreach (var process in organelle.Definition.RunnableProcesses)
             {
                 if (process.Process.Outputs.TryGetValue(oxytoxy, out var oxytoxyAmount))
                 {
                     oxytoxyScore += oxytoxyAmount * Constants.AUTO_EVO_TOXIN_PREDATION_SCORE;
                 }
+            }
+        }
 
-                if (process.Process.Outputs.TryGetValue(mucilage, out var mucilageAmount))
-                {
-                    mucilageScore += mucilageAmount * Constants.AUTO_EVO_MUCILAGE_PREDATION_SCORE;
-                }
+        // Make sure that slime jets are positioned at the back of the cell, because otherwise they will push the cell
+        // backwards (into the predator or away from the prey) or to the side
+        foreach (var slimeJetPosition in slimeJetsHorizontalPositions)
+        {
+            // Range of [0,1] (front to back)
+            var mucilageMultiplier = (float)(slimeJetPosition - topPosition) / (bottomPosition - topPosition);
+            if (mucilageMultiplier <= 0.5)
+            {
+                slimeJetScore *= 0;
+            }
+            else
+            {
+                slimeJetScore += mucilageMultiplier;
```

Comment:
It seems like this logic is dependent on the order in which organelles are traversed. For example, if the first three mucilage were facing forward, but a fourth was facing backwards, it would set the `slimeJetScore` to zero three times, then add the fourth one, right?

I might suggest something like this:
```suggestion
                slimeJetScore += Constants.AUTO_EVO_MUCILAGE_PREDATION_SCORE * (mucilageMultiplier - 0.5f);
```

Reply:
well the problem was with the operation - it was meant to be multiplication instead of addition :sweat_smile: